articles/introduction-to-elastic-stack/index.md

```diff
@@ -0,0 +1,133 @@
+description: This article will cover the basics of Elastic Stack - a log management platform
+author: elly-omondi
+date: 2021-11-08T00:00:00-01:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+
+   - url: /engineering-education/introduction-to-elastic-stack/hero.jpg
+    alt: ELK cover image 
+---
+
+
+[**Elastic stack** (ELK)](https://www.elastic.co/elastic-stack)  is a bundle of software components that can search, visualize, and manage log files(end to end log analysis).
+<!--more-->
+
+Computing devices perform activities and processes that upon completion result into logs that are stored as files. These log data can be  present in servers(i.e. from completed user authentication processes) as well as mobile devices (i.e from activity log).
+The Elastic stack(ELK) provides a log management platform of these log files.
+
+
+
+### Overview
+This article will cover:
+- [What is Elastic Stack](#What-is-elastic-stack)
+- [Components of the Elastic Stack](#Components-of-the-elastic-stack)
+- [How ELK Works](#CHow-ELK-works)
+- [Benefits of using ELK](#Benefits-of-using-ELK)
+
+### What is Elastic Stack
+
+Elastic stack, popularly known as the ELK stack, is a bundle of software tools that provide the ability to search, view and manage log files. The ELK stack processes the logs(in any format) from all systems and applications in use and stores them as a JSON document.
+
+The ELK gives users the ability to analyze these  logs to create meaningful visualisations, perform infrastructure and system monitoring, do efficient troubleshooting and even identify and raise security issues.
+
+The ELK stack is popular because it is an open-source software. Along with this, it is an easily scalable software which makes it the perfect choice for personal use as well as deploying it in your organization. 
+
+There is also a vendor supported paid version of the software suite with added functionality and added cloud support.
+
+For many years, the Elastic Stack existed as a collection of 3 tools,(hence the acronym ELK),which included:
++ Elasticsearch
++ Logstash
++ Kibana
+
+Recently, a fourth component, Beats , was added to the suite by the Elastic company.
+The Elastic Stack is made up of 4 independent, powerful and compatible open source tools. All of the components of the stack can be used as standalone applications as they are compatible with a multitude of other applications.
+
+![Flow diagram](/engineering-education/introduction-to-elastic-stack/Flow-diagram.png)
+
+### Components of the Elastic Stack
+
+1. ### Elasticsearch
+[**Elasticsearch**] (https://www.elastic.co/elasticsearch/) is a distributed search and analytics engine (with apache lucene as the base).This distributed, open-source tool acts as the engine of the entire suite and supports all data types/formats i.e. texts, semi-structured data or even unstructured data. It is essentially a really powerful database where all the logs are stored for other applications to retrieve later. 
```

Comment:
remove space between [elasticsearch] and (https://) for proper markdown formatting for all the 4 bullets